### PR TITLE
Added config for AbuseIPDB, ony tested on Ubuntu 16.04

### PR DESCRIPTION
**Pull request by ajcollett**
Added code from https://wiki.shaunc.com/wikka.php?wakka=ReportingToAbuseIPDBWithFail2Ban

- I assume this should go into 0.10 branch.
- Don't know if this needs tests. Probably not?
- No open issues as far as I can see.
- The ChangeLog will need to be updated, I'd rather leave this to someone who knows the format a bit better than myself. 

I am not sure why this hasn't been added yet, but i am happy to fix anything that needs to be fixed in the PR if that is needed.

### CODE REVIEW

**Comment by mitchellkrogza** [comment]
Nice addition to Fail2Ban. I am about to start testing 0.10 on a testbed before I put it into production will wait until this has been merged before I do. Thanks.

File: config/jail.conf [hunk 1]
```diff
@@ -206,6 +206,23 @@ action_badips = badips.py[category="%(__name__)s", banaction="%(banaction)s", ag
 # Report ban via badips.com (uses action.d/badips.conf for reporting only)
 #
 action_badips_report = badips[category="%(__name__)s", agent="%(fail2ban_agent)s"]
+# Actions to report to abuseipdb.com via API.
+# See action.d/abuseipdb.conf and https://wiki.shaunc.com/wikka.php?wakka=ReportingToAbuseIPDBWithFail2Ban
+# If you want cleaner reports that ensure no user data see the helper script at the above IP
+# NOTE: These reports may include sensitive Info.
+
+# IMPORTANT: Register for abuseipdb [https://www.abuseipdb.com], get, and set the api key at the bottom of action.d/abuseipdb.conf
```
Comment (sebres) [inline config/jail.conf]: This lot of parameters burdened the jail substitution, therefore belong imho into corresponding action...

File: config/jail.conf [hunk 2]
```diff
@@ -330,7 +347,7 @@ logpath = /opt/openhab/logs/request.log
 port    = http,https
 logpath = %(nginx_error_log)s
 
-# To use 'nginx-limit-req' jail you should have `ngx_http_limit_req_module` 
+# To use 'nginx-limit-req' jail you should have `ngx_http_limit_req_module`
```
Comment (sebres) [inline config/jail.conf]: please don't touch lines not affected by this PR (I suspect you've trim ending spaces in your editor, but you can revert such lines before commit, so please rebase this)

Comment (ajcollett) [inline config/jail.conf]: Thanks for this, I saw that before commit and thought "well, it'll probably happen in someones editor eventually, may as well trim them where they come up." But, I am happy to revert those changes.

File: config/jail.conf (hunk 1 above)
Comment (ajcollett) [inline config/jail.conf]: Will probably fix in the next few days. Shouldn't be much work at all.

**Comment by ajcollett** [comment]
As a general thing, are there any thoughts about the content that is sent to abuseipdb.com?

Here is an example submission from a server of mine. (That is Andrew James Collett)
https://abuseipdb.com/check/117.207.208.32

Notice the hostname "backend" is included. Should I put the time in to remove that via a script?
Or leave that to the user?

**Comment by ajcollett** [comment]
@sebres apologies for so many commits. Let me know if there is anything else you'd like changed.

**Comment by sebres** [comment]
Merged... Thanks!

**Comment by ajcollett** [comment]
I'm glad, Pleasure! :-D

**Pull request merged**